quickstarts/microsoft.databasewatcher/watcher-new-datastore/azuredeploy.parameters.json

```diff
@@ -37,7 +37,13 @@
         },
         "targetProperties": {
             "value": [
-                "{\"targetAuthenticationType\":\"Aad\",\"connectionServerName\":\"quickstart-e1a0987a-54ee-46e0-b76a-12f36c835fe0.database.windows.net\",\"targetType\":\"SqlDb\",\"sqlDbResourceId\":\"/subscriptions/3b880560-bc3b-4d17-ab39-d37a1277b841/resourceGroups/quickstart-d2e98952-a48a-4128-b5d5-920f5c64c514/providers/Microsoft.Sql/servers/quickstart-e1a0987a-54ee-46e0-b76a-12f36c835fe0/databases/quickstart-database\",\"readIntent\":false}"
+                {
+                    "targetType": "SqlDb",
+                    "sqlDbResourceId": "/subscriptions/GEN-GUID/resourceGroups/GEN-UNIQUE/providers/Microsoft.Sql/servers/GEN-UNIQUE/databases/GEN-UNIQUE",
```

Comment:
This is not going to work. `GEN-*` can only be a parameter value that you use in the template. So to achieve this you would need something like the following:

```json
"serverName": {
  "value": "GEN-UNIQUE"
},
"targetProperties": {
  "value": {
    "targetType": "SqlDb",
    ...
   "connectionServerName": "[concat(parameters('serverName'), 'database.windows.net')]"
  }
}
```

Separately, you should not be generating subscription GUIDs and resourceGroup names randomly. You should instead use the [`resourceId()`](https://learn.microsoft.com/en-us/azure/azure-resource-manager/templates/template-functions-resource#resourceid) function inside of the template instead of constructing it yourself as a parameter value. The `resourceId()` function will use the subscription and resourceGroup context of the current deployment unless you need to override, but even then it should not be random.